File: TodoApi/Models/Resources/Domain/Resource.cs
using System;
using System.Collections.Generic;
using FrameworkDDD.Common;

namespace TodoApi.Models.Resources
{
    public class Resource : IAggregateRoot
    {
    public string Code { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Status { get; set; } = "Active";
        public decimal OperationalCapacity { get; set; }
        public string? AssignedArea { get; set; }
        public int? SetupTimeMinutes { get; set; }
    public List<ResourceQualification> RequiredQualifications { get; private set; }

        public Resource()
        {
            RequiredQualifications = new List<ResourceQualification>();
        }
    }
}

File: frontend/src/app/models/resource.ts
export interface ResourceDTO {
  code: string;
  description: string;
  type: string;
  status: string;
  operationalCapacity: number;
  assignedArea?: string | null;
  setupTimeMinutes?: number | null;
  requiredQualifications: string[];
}

export interface CreateResourceDTO {
  code: string;
  description: string;
  type: string;
  operationalCapacity: number;
  status: string;
  assignedArea?: string | null;
  setupTimeMinutes?: number | null;
  requiredQualifications: string[];
}

export interface UpdateResourceDTO {
  description: string;
  type: string;
  operationalCapacity: number;
  status: string;
  assignedArea?: string | null;
  setupTimeMinutes?: number | null;
  requiredQualifications: string[];
}
